include/migration/vmstate.h

```diff
@@ -1084,17 +1098,44 @@ extern const VMStateInfo vmstate_info_qlist;
 #define VMSTATE_UINT64_ARRAY(_f, _s, _n)                              \
     VMSTATE_UINT64_ARRAY_V(_f, _s, _n, 0)
 
-#define VMSTATE_CAP_ARRAY_V(_f, _s, _n, _v)                                    \
-    VMSTATE_ARRAY(_f, _s, _n, _v, vmstate_info_cap_register, cap_register_t)
+#define VMSTATE_CAP64_ARRAY_V(_f, _s, _n, _v)                         \
+    VMSTATE_ARRAY(_f, _s, _n, _v, vmstate_info_cap64_register, cc64_cap_t)
+
+#define VMSTATE_CAP64_ARRAY(_f, _s, _n)                               \
+    VMSTATE_CAP64_ARRAY_V(_f, _s, _n, 0)
+
+#define VMSTATE_CAP128_ARRAY_V(_f, _s, _n, _v)                        \
+    VMSTATE_ARRAY(_f, _s, _n, _v, vmstate_info_cap128_register, cc128_cap_t)
+
+#define VMSTATE_CAP128_ARRAY(_f, _s, _n)                              \
+    VMSTATE_CAP128_ARRAY_V(_f, _s, _n, 0)
+
+#define VMSTATE_CAP128M_ARRAY_V(_f, _s, _n, _v)                       \
+    VMSTATE_ARRAY(_f, _s, _n, _v, vmstate_info_cap128m_register, cc128m_cap_t)
+
+#define VMSTATE_CAP128M_ARRAY(_f, _s, _n)                             \
+    VMSTATE_CAP128M_ARRAY_V(_f, _s, _n, 0)
+
+#define VMSTATE_ALIGN_CAP64_ARRAY_V(_f, _s, _n, _v)                   \
```

Comment:
Do we need both align and non-align versions of this?

Reply:
We don't really do vmstate at the moment, but we'd want unaligned ones for CHERI-RISC-V (as we don't align for TCG things) but aligned ones for Morello with the current implementations of them. Plus both already existed, this is just expanding out what was already there.